From 8ed64d11f31c76275b335632ca14bfe924d7aca0 Mon Sep 17 00:00:00 2001
From: Jon Mason <jon.mason@arm.com>
Date: Sat, 17 Aug 2024 15:02:10 -0400
Subject: [PATCH] Update include file for 6.10 and later kernels

In the Linux kernel v6.10 kernel, the tee include file tee_drv.h was
modified and some portions were split into a new file, tee_core.h.
Kernel commit 0439fcff304acdedfc493dbf900dce86922fd31f
Update the references here to point to the new file, with some #if's
around it to help with kernel version compatibility.

Signed-off-by: Jon Mason <jon.mason@arm.com>
---
 core.c          | 4 ++++
 shm_pool.c      | 4 ++++
 tstee_private.h | 4 ++++
 3 files changed, 12 insertions(+)

diff --git a/core.c b/core.c
index 23a05043995a..e1afff060c10 100644
--- a/core.c
+++ b/core.c
@@ -18,6 +18,10 @@
 #include <linux/slab.h>
 #include <linux/stat.h>
 #include <linux/tee_drv.h>
+#include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,10,0)
+#include <linux/tee_core.h>
+#endif
 #include <linux/types.h>
 #include <linux/uaccess.h>
 
diff --git a/shm_pool.c b/shm_pool.c
index d9dd76adead7..9ea5f66b598d 100644
--- a/shm_pool.c
+++ b/shm_pool.c
@@ -9,6 +9,10 @@
 #include <linux/genalloc.h>
 #include <linux/slab.h>
 #include <linux/tee_drv.h>
+#include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,10,0)
+#include <linux/tee_core.h>
+#endif
 
 #include "tstee_private.h"
 
diff --git a/tstee_private.h b/tstee_private.h
index e54027843f2d..d7b2ba9dbab1 100644
--- a/tstee_private.h
+++ b/tstee_private.h
@@ -10,6 +10,10 @@
 #include <linux/bitops.h>
 #include <linux/idr.h>
 #include <linux/tee_drv.h>
+#include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,10,0)
+#include <linux/tee_core.h>
+#endif
 #include <linux/types.h>
 #include <linux/uuid.h>
 
-- 
2.39.2

